pkg/executor/join/anti_semi_join_probe.go

```diff
@@ -0,0 +1,329 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package join
+
+import (
+	"unsafe"
+
+	"github.com/pingcap/tidb/pkg/expression"
+	"github.com/pingcap/tidb/pkg/util/chunk"
+	"github.com/pingcap/tidb/pkg/util/sqlkiller"
+)
+
+type antiSemiJoinProbe struct {
+	baseSemiJoin
+
+	// After processing all probe rows, we will find which rows will be outputted
+	isMatchedRowsAdjustedForOutput bool
+}
+
+func newAntiSemiJoinProbe(base baseJoinProbe, isLeftSideBuild bool) *antiSemiJoinProbe {
+	ret := &antiSemiJoinProbe{
+		baseSemiJoin: *newBaseSemiJoin(base, isLeftSideBuild),
+	}
+
+	if ret.ctx.hasOtherCondition() {
+		ret.isNulls = make([]bool, 0, 32)
```

Comment:
should use `chunk.InitialCapacity` instead of `32`

Reply:
> should use `chunk.InitialCapacity` instead of `32`

done